cmd/tendermint/commands/inspect.go

```diff
@@ -0,0 +1,87 @@
+package commands
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/inspect"
+	"github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/state/indexer/block"
+	"github.com/tendermint/tendermint/store"
+	"github.com/tendermint/tendermint/types"
+)
+
+// InspectCmd is the command for starting an inspect server.
+var InspectCmd = &cobra.Command{
+	Use:   "inspect",
+	Short: "Run an inspect server for investigating Tendermint state",
+	Long: `
+	inspect runs a subset of Tendermint's RPC endpoints that are useful for debugging
+	issues with Tendermint.
+
+	When the Tendermint consensus engine detects inconsistent state, it will crash the
+	tendermint process. Tendermint will not start up while in this inconsistent state. 
+	The inspect command can be used to query the block and state store using Tendermint
+	RPC calls to debug issues of inconsistent state.
+	`,
+
+	RunE: runInspect,
+}
+
+func init() {
+	InspectCmd.Flags().
+		String("rpc.laddr",
+			config.RPC.ListenAddress, "RPC listenener address. Port required")
+	InspectCmd.Flags().
+		String("db-backend",
+			config.DBBackend, "database backend: goleveldb | cleveldb | boltdb | rocksdb | badgerdb")
+	InspectCmd.Flags().
+		String("db-dir", config.DBPath, "database directory")
+}
+
+func runInspect(cmd *cobra.Command, args []string) error {
```

Comment:
Will not closing any of the stores if an error occurs result in database corruption in any way, or is it safe to do so? The only place the block store DB's cleanly closed when an error occurs in this function is on line 65.

If it could end up being problematic, this function seems like a good candidate for applying [this kind of approach](https://github.com/tendermint/tendermint/pull/9463/files#diff-99ad517370fe6ad4d3e2be420c6845234709b5c96326e6cc3ad635d84fdff8a4R82-R87).

Reply:
I think almost all databases will have crash protection semantics so there should be no risk of database corruption (actually `Close` for some DBs that Tendermint supports is a noop). Also inspect is read-only. 

Nevertheless probably still good practice to add the defer statements

Reply:
Added defers.